### PR TITLE
[Doc] Change the description for pip packages

### PR DESCRIPTION
**Pull request by TaoLv**
## Description ##

- Remove experimental for MKL-DNN backend.
- Add explaination for pip package name.

@aaronmarkham @pengzhao-intel

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by stu1130** [comment]
Thanks for the contribution @TaoLv
@mxnet-label-bot[pr-awaiting-review]

File: docs/install/index.md [hunk 1]
```diff
@@ -141,11 +141,17 @@ $ pip install mxnet --pre
...
+MXNet offers MKL pip packages that will be much faster when running on Intel hardware.
+Check the chart below for other options, refer to <a href="https://pypi.org/project/mxnet/">PyPI for other MXNet pip packages</a>, or <a href="validate_mxnet.html">validate your MXNet installation</a>. 
 
 <img src="https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/install/pip-packages-1.3.0.png" alt="pip packages"/>
 
+**NOTES:**
+
+*mxnet-cu92mkl* means the package is built with CUDA/cuDNN and MKL-DNN enabled and the CUDA version is 9.2.
+
+MKL pip package is experimental in some old versions of MXNet.
```
Comment (aaronmarkham) [inline docs/install/index.md]: Wouldn't it be more accurate to say "All MKL pip packages are experimental prior to version x.y.z."

Comment (TaoLv) [reply docs/install/index.md]: Thanks for the clarification. Will change the doc.

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy  Please change label to pr-awaiting-merge. This PR has been approved.

**Pull request merged**